Feature/main page에서 master로 머지 요청.

donghk0128 opened the pull request:
현재 DB, 아이콘 추천 기능이 어느 정도 구현이 되었으나, main-page branch에서의 작업은 master branch에 update된 기능을 배제하고 있는 상황입니다. 또한 part 3, 4가 main page의 기능을 필요로 하는 경우가 생길 것을 고려하여 main-page branch를 master에 머지 후 새롭게 branch를 팔 것을 제안합니다.

CaChiJ committed:
feature : 사용자의 터치에 따라 회전하는 메인다이얼

메인페이지의 메인다이얼을 추가함.
메인다이얼이 사용자의 터치와 드래그에 따라 회전하는 기능을 추가함.

donghk0128 committed:
feat : 임박한 다이얼 표시 기능 추가

임박한 다이얼 기능 구현을 위해 클래스를 추가함.

donghk0128 committed:
feat : 임박한 다이얼 표시 기능

임박한 다이얼 기능 구현 코드 작성.

CaChiJ committed:
chore : 필요 없는 아이콘 제거

이전에 실수로 추가했던 아이콘을 제거함.

CaChiJ committed:
chore : SpinnableImageView 회전 시간 옵션 추가

SpinnableImageView의 rotate함수에 duration 매개변수를 추가함. 이 매개변수에 전달된 인자를 통해 회전 시간을 지정할 수 있음.
덧붙여, 관련 함수(reset, rotate)의 매개변수 형에 final을 추가함.

CaChiJ committed:
feat : 메인다이얼 자석 기능

메인 다이얼을 조작한 뒤에 가장 가까운 카테고리의 각도로 메인 다이얼이 회전함.
메인 다이얼의 각도를 [0, 360)으로 제한함.
+ 주석과 매개변수의 final 키워드를 추가함.

CaChiJ committed:
feat : 메인 다이얼 카테고리 선택 시 진동

메인 다이얼을 조작해 카테고리를 선택하면 진동을 울리게 함.
+ 불필요한 assert문 제거

CaChiJ committed:
chore : 메인 다이얼 진동 강도 조절

기존보다 진동 세기를 줄였습니다.

donghk0128 committed:
feat : 임박한 다이얼 표시 기능

임박한 다이얼 기능 구현 코드 작성

CaChiJ committed:
refactor : UrgentDials 클래스 구조 수정 및 버그 제거

UrgentDials 클래스를 UrgentDialLayout으로 변경함
남은 시간을 구하는 부분의 버그를 수정함

donghk0128 committed:
test : 임박한 다이얼 표시 기능

테스트 중인 코드임.

donghk0128 committed:
test : 임박한 다이얼 표시 기능

테스트 중인 코드임.

CaChiJ committed:
feat : 임박다이얼 표시 (레이아웃 부분)

메인페이지에서 임박다이얼을 동적으로 표시하는 레이아웃을 생성함.
임박다이얼 표시 포맷은 수정이 필요함.
UrgentDialAdapter의 생성자가 테스트 코드를 포함하고 있으므로 추후에 제거해야 함.

CaChiJ committed:
feat : 메인페이지 임박다이얼 이름 표시

메인페이지의 임박다이얼의 이름이 표시되도록 수정하였습니다.

CaChiJ committed:
feat : 메인 다이얼 다이얼 매니저 연동

메인 다이얼의 카테고리 원이 다이얼 매니저의 카테고리 개수와 연동됨
현재 선택한 카테고리 번호를 반환하는 함수도 추가함

donghk0128 committed:
feat : 임박한 다이얼 표시 및 카테고리 내 다이얼 표시 기능

donghk0128 committed:
chore : View 변수 선언 수정

메인 Activity 내 View 변수 3개 멤버 변수 제거 및 수정

donghk0128 committed:
chore : View 변수 및 adapter 변수 이름 수정

메인 Activity 내 View 변수 3개 멤버 변수 및 gridAdapter 이름을 camel 표기법에 맞게 수정

donghk0128 committed:
Merge pull request #3 from ProjectFRIDAY/feature/dir-main-page/urgent-dial

Feature/dir main page/urgent dial

donghk0128 committed:
chore : CategoryDialAdapter 코드 수정

빈 카테고리일 시 빈 리스트 보여주도록 코드 수정

CaChiJ committed:
chore : 병합 전 코드 정리

CaChiJ committed:
Merge branch 'feature/main-page' into feature/dir-main-page/main-dial

CaChiJ committed:
chore : 코딩 스타일 개선

빈 칸, 불필요한 명시적 형변환, 불필요한 import를 수정했습니다.

CaChiJ committed:
Merge pull request #6 from ProjectFRIDAY/feature/dir-main-page/main-dial

main-dial 브랜치를 main-page 브랜치에 병합합니다.

DongYeon-Handsome committed:
Feat : Add FrontEnd

donghk0128 committed:
chore : 프론트엔드 수정

donghk0128 committed:
chore : 메인 액티비티 xml 수정

모서리, 위치 설정 변경

donghk0128 committed:
chore : 메인 액티비티 xml 수정

모서리, 위치 설정 변경

CaChiJ committed:
feat : 상단 상태바(status bar)를 투명으로 만듦

상단 상태바를 라이트 모드와 다크 모드 모두에서 투명으로 만들었습니다.

donghk0128 committed:
chore : 카테고리 다이얼 어댑터 수정

메인 다이얼 연동 과정

CaChiJ committed:
chore : 카테고리 선택 시 카테고리 다이얼 어댑터로 값 전달

메인 다이얼로 카테고리를 선택할 때마다 CategoryDialAdapter의 setCategory 메서드를 통해 선택된 카테고리 정보를 전달하도록 함.

CaChiJ committed:
Merge branch 'feature/main-page' into FrontEnd_DY

CaChiJ committed:
chore : 뷰 id 값 변경

CaChiJ committed:
chore : 뷰 id 값 변경

activity_main.xml 파일에 속한 뷰들의 id 값을 feature/main-page의 뷰들과 같게 만듦

CaChiJ committed:
Merge pull request #7 from ProjectFRIDAY/FrontEnd_DY

ui 병합이므로 별도로 리뷰를 받지 않고 병합합니다.

CaChiJ committed:
chore : 잘못된 리소스명 수정

병합중 잘못 수정된 리소스명을 수정합니다.

CaChiJ committed:
chore : 잘못된 xml 파일 수정

ui 병합중 잘못 수정된 xml 태그를 수정하였습니다.

CaChiJ committed:
chore : 메인 페이지 ui 마진 변경

Mango-Juice committed:
feat : 푸시 알림 추가

사용자가 정한 시작일과 주기에 맞춰 푸시 알림을 주는 기능을 추가함.

CaChiJ committed:
feat : 다이얼 갱신, 선택한 다이얼 표시

메인 다이얼 중앙에 클릭한 다이얼을 표시합니다.

Mango-Juice committed:
feat : 긴급 다이얼 목록 자동 동기화

메인 페이지의 긴급 다이얼 목록이 주기적(임의로 1분 간격으로 설정함)으로 동기화될 수 있도록 함.

c-peace committed:
feat : 전원을 껏다 킨 후 다이얼 작동

전원을 껏다 킨 후 다이얼 작동 코드 작성

Mango-Juice committed:
fix : 여러 다이얼을 등록해도 알림이 한 번만 울리는 오류 수정

다이얼을 여러개 등록해도 알림이 한 번만 울리는 오류를 수정하였음.
+ 메인액티비티 테스트 코드 수정

Mango-Juice committed:
Merge pull request #8 from ProjectFRIDAY/feature/dir-main-page/sync-urgent-dial

긴급 다이얼 목록 자동 동기화

Mango-Juice committed:
chore : 푸시 알림시 null이 뜨지 않도록 수정

다이얼 이름이 null로 전달된 경우 아예 알림을 울리지 않도록 수정함.

Mango-Juice committed:
feat : 푸시 알림 기능 완성

기존 푸시 알림 기능이 한 번만 작동하는 문제 수정.
푸시 알림을 눌렀을 때 다시 알림이 뜨는 오류 수정.

Mango-Juice committed:
Merge branch 'feature/main-page' into feature/dir-main-page/background

Mango-Juice committed:
Merge pull request #9 from ProjectFRIDAY/feature/dir-main-page/background

푸시 알림 기능 추가

CaChiJ committed:
chore : warning 제거

자잘한 warning들을 제거했습니다.

CaChiJ committed:
feat : 카테고리 이름 표시

메인 다이얼로 지정한 카테고리의 이름이 표시됩니다.

CaChiJ committed:
chore : 병합

로컬 저장소에서 작업중이던 코드와 충돌이 생겨 직접 병합함.

CaChiJ committed:
chore : 병합 오류 수정

Dial 클래스에서 잘못 병합된 부분을 수정함

donghk0128 committed:
Merge branch 'master' into feature/main-page

Mango-Juice commented:
커밋 53개가 쌓인 거 보니 머지가 필요할 거 같긴 하네요.
다만 머지 전에 main-page 브랜치에서 작업 중이신 분들 다 커밋해주셔야 안전하게 머지할 수 있을 것 같아요!!

donghk0128 committed:
chore : 카테고리 다이얼 갱신 문제 해결

빈 카테고리로 다이얼을 회전시킬 경우 리사이클러뷰 갱신이 안되는 문제 해결

CaChiJ committed:
chore : warning 제거

Category, CategoeryDialAdapter, Dial, DialManager의 Warning을 제거함.

donghk0128 committed:
chore : warning 사항 일부 수정

donghk0128 committed:
chore : warning 사항 일부 수정

CaChiJ committed:
chore : 잘못된 NonNull 어노테이션 수정

androidx.annotation.NonNull이 아니라 com.example.plandial.NonNull.java를 사용하던 부분을 수정했습니다.

Mango-Juice commented:
LGTM 👍👍
